pkg/actions/cluster/owned_test.go

```diff
@@ -98,13 +110,127 @@ var _ = Describe("Delete", func() {
+
+			c.SetNewClientSet(func() (kubernetes.Interface, error) {
+				return fakeClientSet, nil
+			})
+
+			mockedDrainer := &drainerMockOwned{}
+			mockedDrainer.On("Drain", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(errors.New("Mocked error"))
+			c.SetNewNodeGroupManager(func(cfg *api.ClusterConfig, ctl *eks.ClusterProvider, clientSet kubernetes.Interface) cluster.NodeGroupDrainer {
+				return mockedDrainer
+			})
+
+			err := c.Delete(time.Microsecond, false, true, false)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(fakeStackManager.DeleteTasksForDeprecatedStacksCallCount()).To(Equal(1))
+			Expect(ranDeleteDeprecatedTasks).To(BeFalse())
+			Expect(fakeStackManager.NewTasksToDeleteClusterWithNodeGroupsCallCount()).To(Equal(1))
+			Expect(ranDeleteClusterTasks).To(BeFalse())
+			mockedDrainer.AssertNumberOfCalls(GinkgoT(), "Drain", 1)
+		})
+
+		It("nodes draining error thrown when force flag not used", func() {
+			//mocks are in order of being called
+			p.MockEKS().On("DescribeCluster", mock.MatchedBy(func(input *awseks.DescribeClusterInput) bool {
+				Expect(*input.Name).To(Equal(clusterName))
+				return true
+			})).Return(&awseks.DescribeClusterOutput{
+				Cluster: testutils.NewFakeCluster(clusterName, awseks.ClusterStatusActive),
+			}, nil)
+
+			p.MockEKS().On("ListFargateProfiles", &awseks.ListFargateProfilesInput{
+				ClusterName: strings.Pointer(clusterName),
+			}).Once().Return(&awseks.ListFargateProfilesOutput{FargateProfileNames: aws.StringSlice([]string{})}, nil)
+
+			p.MockEKS().On("DeleteFargateProfile", &awseks.DeleteFargateProfileInput{
+				ClusterName:        aws.String(clusterName),
+				FargateProfileName: aws.String("fargate-1"),
+			}).Once().Return(&awseks.DeleteFargateProfileOutput{}, nil)
+
+			p.MockEKS().On("ListFargateProfiles", &awseks.ListFargateProfilesInput{
+				ClusterName: strings.Pointer(clusterName),
+			}).Once().Return(&awseks.ListFargateProfilesOutput{}, nil)
+
+			fakeStackManager.DeleteTasksForDeprecatedStacksReturns(&tasks.TaskTree{
+				Tasks: []tasks.Task{},
+			}, nil)
+			fakeStackManager.ListNodeGroupStacksReturns([]manager.NodeGroupStack{{NodeGroupName: "ng-1"}}, nil)
+
+			p.MockEC2().On("DescribeKeyPairs", mock.Anything).Return(&ec2.DescribeKeyPairsOutput{}, nil)
+
+			p.MockEC2().On("DescribeSecurityGroupsWithContext", mock.Anything, mock.Anything).Return(&ec2.DescribeSecurityGroupsOutput{}, nil)
+
+			fakeStackManager.NewTasksToDeleteClusterWithNodeGroupsReturns(&tasks.TaskTree{
+				Tasks: []tasks.Task{},
+			}, nil)
+
+			c := cluster.NewOwnedCluster(cfg, ctl, nil, fakeStackManager)
+			fakeClientSet = fake.NewSimpleClientset()
+
+			c.SetNewClientSet(func() (kubernetes.Interface, error) {
+				return fakeClientSet, nil
+			})
+
+			mockedDrainer := &drainerMockOwned{}
+			mockedDrainer.On("Drain", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(errors.New("Mocked error"))
```

Comment:
rather than `mock.Anything` can we replace it with the arguments we expect (for some at least)? Makes the test more explicit

Reply:
same goes for a few of the other new tests

Reply:
Done